### DISCUSSION TITLE
Tasmota scripting with Sonoff POW R3

### DISCUSSION THREAD

**Discussion by yves-jose** [arendst/Tasmota#17927] [Ideas]
I made an electric vehicle charger in Tasmota Scriting on a Wemos d1 Mini (which works well). I would like it installed on a Sonoff Pow R3 but Tasmota Scripting (repository firmware @jason2866) does not have it.
Can you help me to realize this project.
Note: I use google translate because my English is very bad.

**Comment by barbudor** [comment]
I don't understand your problem
Sonoff Pow R3 uses an ESP8266 just like the Wemos D1 mini
So the same firmware should be used : tasmota-scripting.bin(.gz) from here https://github.com/Jason2866/Tasmota-specials/tree/firmware/firmware/tasmota/other

Or use the tasmota web-installer : https://tasmota.github.io/install/
And choose "Tasmota Scripting (english)" in the list (close to the end)

![image](https://user-images.githubusercontent.com/2996042/217940929-6b9778cc-0164-4da6-a94c-f463ce07b540.png)


Then don't forget to set the template from https://templates.blakadder.com/sonoff_POWR3.html

**Reply by yves-jose** [reply]
Tasmota scripting does not integrate CSE7766 Tx and CSE7766 Rx to be able to use Sonoff Pow 3. Set the template from https://templates.blakadder.com/sonoff_POWR3.html 
does not work the GPIO01 and GPIO03 field remains blank

**Reply by barbudor** [reply]
This message is clearer than the 1st one
You could have build yourself a tasmota that match your need with TasmoCompiler

**Comment by Jason2866** [comment]
@yves-jose Added with this https://github.com/Jason2866/Tasmota-build/commit/9ddacf893fae3d81242dd755b7a41140ffd480cf 
with next commit from Theo the specials will be build.
The energy monitoring should work with the Pow. Hope that the increase makes the variant not too big.

**Reply by yves-jose** [reply]
Thank you for your prompt response. I'm waiting for this special build.

**Reply by ThinkPadNL** [reply]
I just flashed a `tasmota-scripting.bin.gz` that was built today. Unfortunately no power data is shown on my Sonoff POW R2.
Probably the support for power measures has been removed again from the scripting build.
I just made my own build by running `TasmoCompiler` in a Docker image. Was very easy actually, handy GUI that lets you select the components you want included. Works great now.

I use the self compiled firmware on a Sonoff POW R2 that i use to monitor my solar panels upload. I needed the scripting functionality so that the device can upload the generation data to PVOutput.org by itself.

**Comment by Jason2866** [comment]
Had to remove the energy monitoring drivers. The firmware size with enabled energy drivers is to big for OTA updates.
